File: ProductMicroservice/Models/CartItems.cs
using ProductMicroservice.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ProductMicroservice.Models
{
    public class CartItems
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int CartItemId { get; set; }
        public int CartId { get; set; }
        public int ProductId { get; set; }

        [ForeignKey("CartId")]
        public Carts Carts { get; set; }

        [ForeignKey("ProductId")]
        public Products Products { get; set; }
        public int Quantity { get; set; }
    }
}

File: Project Main Ui/src/app/Models/CartItems.ts
interface Product {
    productId: number;
    name: string;
    description: string;
    categoryId: number;
    price: number;
    imageName: string;
    category: any;
  }
  
  interface CartItem {
    cartItemId: number;
    cartId: number;
    productId: number;
    carts: any;
    products: Product;
    quantity: number;
  }
  
export interface Carts {
    cartId: number;
    userId: number;
    totalPrice: number;
    users: any;
    cartItems: CartItem[];
  }
  